core/computed/resource-summary.js

```diff
@@ -32,14 +32,42 @@ class ResourceSummary {
     return requestToResourceType[record.resourceType] || 'other';
   }
 
+  /**
+   * Although the browser will only need one favicon asset, we don't know which
+   * one will actually be requested. The browser could request multiple if initial
+   * candidates are found to be unusable. This function simply returns the set of all
+   * possible favicon urls.
+   * @see https://html.spec.whatwg.org/multipage/links.html#linkTypes
+   * @param {LH.Artifacts.LinkElement[]} LinkElements
+   * @param {string|undefined} mainDocumentUrl
+   * @return {string[]}
+   */
+  static determineFaviconUrls(LinkElements, mainDocumentUrl) {
+    if (!mainDocumentUrl) return [];
+
+    const iconLinkElements =
+      LinkElements.filter(e => e.rel === 'icon' || e.rel === 'shortcut icon');
+    if (!iconLinkElements.length) return [new URL('/favicon.ico', mainDocumentUrl).href];
+
+    const urls = [];
+    for (const linkElement of iconLinkElements) {
+      if (linkElement.href) {
+        urls.push(new URL(linkElement.href, mainDocumentUrl).href);
```

Comment:
because that's the url the href of the link element is resolved against.

Reply:
Is it? When I create a link with a relative URL, JS will always return the fully resolved `href`

![Screen Shot 2023-01-27 at 4 32 26 PM](https://user-images.githubusercontent.com/6752989/215230112-9e444a32-4df9-40c6-bd95-004b3c9df7e8.png)

Reply:
ah, good point here. and when from a header we resolve in the gather, so this href is always null or an absolute url....ok!

Reply:
what about the favicon one? I noticed the link element gather uses `finalDisplayedUrl` to resolve headers.... is that really right?

Reply:
> is that really right?

I think so. Using `finalDisplayedUrl` as the base is fine because we aren't using the url path and hash from `finalDisplayedUrl`. It isn't possible to change the domain via history api as far as I know.